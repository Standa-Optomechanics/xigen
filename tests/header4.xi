protocol "v1"
defaults with crc, answer, public

command "foo1_settings" universal "fo1" (10) 
fields:
	int32u speed			

/** $XIR
 * reader
 */
/** $XIW
 * writer
 */
/** $XIS
 * struct
 */
command "foobar" universal "fbr" (10)
with dualsync
fields:
  int32s position

command "bazbaz" reader "rbbz" (10)
with dualsync
fields:
  int32s velocity

command "quxqux_impl" reader "rqqx" (10)
with dualsync
fields:
  int32s velocity

command "foo2_settings" universal "fo2" (10) 
fields:
	int32u speed			

