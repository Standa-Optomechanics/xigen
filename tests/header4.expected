
/** @cond DO_NOT_WANT */
#define LIBXIMC_PROTOCOL_VERSION 1
/** @endcond */


/*
 ------------------------------------------
   BEGIN OF GENERATED struct declarations
 ------------------------------------------
*/

	typedef struct
	{
		unsigned int speed;
	} foo1_settings_t;

/** 
 * struct
	* @see get_foobar, set_foobar
 */
	typedef struct
	{
		int position;
	} foobar_t;

	typedef struct
	{
		int velocity;
	} bazbaz_t;

	typedef struct
	{
		int velocity;
	} quxqux_t;

	typedef struct
	{
		unsigned int speed;
	} foo2_settings_t;

/*
 --------------------------------------------
   BEGIN OF GENERATED function declarations
 --------------------------------------------
*/
	result_t XIMC_API set_foo1_settings (device_t id, const foo1_settings_t* foo1_settings);

	result_t XIMC_API get_foo1_settings (device_t id, foo1_settings_t* foo1_settings);

	result_t XIMC_API set_foobar_unsynced (device_t id, const foobar_t* foobar);

/** 
 * writer
 */
	result_t XIMC_API set_foobar (device_t id, const foobar_t* foobar);

	result_t XIMC_API get_foobar_unsynced (device_t id, foobar_t* foobar);

/** 
 * reader
 */
	result_t XIMC_API get_foobar (device_t id, foobar_t* foobar);

	result_t XIMC_API get_bazbaz_unsynced (device_t id, bazbaz_t* bazbaz);

	result_t XIMC_API get_bazbaz (device_t id, bazbaz_t* bazbaz);

	result_t XIMC_API get_quxqux_impl_unsynced (device_t id, quxqux_t* quxqux);

	result_t XIMC_API get_quxqux_impl (device_t id, quxqux_t* quxqux);

	result_t XIMC_API set_foo2_settings (device_t id, const foo2_settings_t* foo2_settings);

	result_t XIMC_API get_foo2_settings (device_t id, foo2_settings_t* foo2_settings);


/*
 -------------------------
   END OF GENERATED CODE
 -------------------------
*/
